### PR TITLE
Summarize population by age for population projection

### PR DESCRIPTION
**Pull request by maggiesu0725**

### CODE REVIEW

**Comment by maggiesu0725** [comment]
@lhendey please review

File: Prog/reshape ACS data for population projection.do [hunk 1]
```diff
@@ -0,0 +1,568 @@
+********************************************************************************
+* Author: Yipeng Su
+* Project: Regional Housing Framework
+* Date: 12/28/2018
+* Moderation:
+********************************************************************************
+
+import excel "L:\Libraries\RegHsg\Raw\ACS 1 year estimate\All_1year.xlsx", sheet("Sheet1") firstrow
+
+label variable HC01_EST_VC01 "totpop"
```
Comment (lhendey) [inline Prog/reshape ACS data for population projection.do]: these labels should be written out especially as they mirror the vars created below. you can use the metadata files from the aff download to make this easier.

File: Prog/reshape ACS data for population projection.do [hunk 2]
```diff
@@ -0,0 +1,568 @@
...
+			   
+
+
+save "D:\DCData\Libraries\RegHsg\Prog\08-16 ACS 1year.dta"
+
+clear
+
+import excel "L:\Libraries\RegHsg\Raw\ACS 1 year estimate\ACS_17_1YR_2017.xlsx", sheet("Sheet1") firstrow
+
+label variable HC01_EST_VC01 "totpop"
```
Comment (lhendey) [inline Prog/reshape ACS data for population projection.do]: revise labels as above

File: Prog/reshape ACS data for population projection.do [hunk 3]
```diff
@@ -0,0 +1,568 @@
...
+rename HC05_EST_VC01 HC03_EST_VC01
+			   
+append using "D:\DCData\Libraries\RegHsg\Prog\08-16 ACS 1year.dta", force
+
+sort Year
+
+rename HC01_EST_VC01 totpop
+rename HC02_EST_VC01 totmalepop
+rename HC03_EST_VC01 totfemalepop
+
```
Comment (lhendey) [inline Prog/reshape ACS data for population projection.do]: add labels for new variables created.

File: Prog/reshape ACS data for population projection.do [hunk 4]
```diff
@@ -0,0 +1,568 @@
...
+export excel using "D:\DCData\Libraries\RegHsg\Prog\ACS_1year_gender.xlsx"
+
+clear
+
+***************************population by gender 5 year estimate ****************
+
+import excel "L:\Libraries\RegHsg\Raw\ACS 5 year estimate\All_5year.xlsx", sheet("Sheet1") firstrow
+
+label variable HC01_EST_VC01 "totpop"
+label variable HC02_EST_VC01 "mtotpop"
```
Comment (lhendey) [inline Prog/reshape ACS data for population projection.do]: i don't think these are right. hc02_est_vc01 is the percent of the total population according to the metadata.  file:///L:\Libraries\RegHsg\Raw\ACS%201%20year%20estimate\aff_download\ACS_17_1YR_S0101_metadata.csv

File: Prog/reshape ACS data for population projection.do [hunk 5]
```diff
@@ -0,0 +1,568 @@
...
+label variable HC01_EST_VC19 "tot80to84"
+label variable HC02_EST_VC19 "mtot80to84"
+label variable HC03_EST_VC19 "ftot80to84"
+label variable HC01_EST_VC20 "tot85up"
+label variable HC02_EST_VC20 "mtot85up"
+label variable HC03_EST_VC20 "ftot85up"
+
+
+gen totunder5= HC01_EST_VC03*HC01_EST_VC01/100
+gen mtotunder5= HC02_EST_VC03*HC01_EST_VC01/100
```
Comment (lhendey) [inline Prog/reshape ACS data for population projection.do]: i don't think these are all right. HC02_EST_VC03 is percent of kids under 5 not number of males under 5 according to the metadata.

File: Prog/reshape ACS data for population projection.do [hunk 6]
```diff
@@ -0,0 +1,568 @@
...
+collapse (sum) HC01_EST_VC01 HC02_EST_VC01 HC03_EST_VC01 totunder5 mtotunder5 ftotunder5 tot5to9 mtot5to9 ftot5to9 tot10to14 mtot10to14 ftot10to14 ///
+              tot15to19  mtot15to19 ftot15to19 tot20to24 mtot20to24 ftot20to24 tot25to29 mtot25to29 ftot25to29 ///
+			  tot30to34  mtot30to34 ftot30to34 tot35to39 mtot35to39 ftot35to39 tot40to44 mtot40to44 ftot40to44 ///
+			   tot45to49  mtot45to49 ftot45to49 tot50to54 mtot50to54 ftot50to54 tot55to59 mtot55to59 ftot55to59 ///
+			   tot60to64 mtot60to64 ftot60to64 tot65to69 mtot65to69 ftot65to69 tot70to74 mtot70to74 ftot70to74 ///
+			   tot75to79 mtot75to79 ftot75to79 tot80to84 mtot80to84 ftot80to84 tot85up mtot85up ftot85up, by (Year)
+			   
+			   
+			  
+save "D:\DCData\Libraries\RegHsg\Prog\09-16 ACS 5year.dta"
```
Comment (lhendey) [inline Prog/reshape ACS data for population projection.do]: save to the data folder not prog.

File: Prog/reshape ACS data for population projection.do [hunk 7]
```diff
@@ -0,0 +1,568 @@
...
+collapse (sum) HC01_EST_VC01 HC02_EST_VC01 HC03_EST_VC01 totunder5 mtotunder5 ftotunder5 tot5to9 mtot5to9 ftot5to9 tot10to14 mtot10to14 ftot10to14 ///
+              tot15to19  mtot15to19 ftot15to19 tot20to24 mtot20to24 ftot20to24 tot25to29 mtot25to29 ftot25to29 ///
+			  tot30to34  mtot30to34 ftot30to34 tot35to39 mtot35to39 ftot35to39 tot40to44 mtot40to44 ftot40to44 ///
+			   tot45to49  mtot45to49 ftot45to49 tot50to54 mtot50to54 ftot50to54 tot55to59 mtot55to59 ftot55to59 ///
+			   tot60to64 mtot60to64 ftot60to64 tot65to69 mtot65to69 ftot65to69 tot70to74 mtot70to74 ftot70to74 ///
+			   tot75to79 mtot75to79 ftot75to79 tot80to84 mtot80to84 ftot80to84 tot85up mtot85up ftot85up, by (Year)
+			   
+
+
+save "D:\DCData\Libraries\RegHsg\Prog\08-16 ACS 1year.dta"
```
Comment (lhendey) [inline Prog/reshape ACS data for population projection.do]: save to the data folder not prog.

File: Prog/reshape ACS data for population projection.do [hunk 8]
```diff
@@ -0,0 +1,568 @@
...
+
+rename HC01_EST_VC01 totpop
+rename HC02_EST_VC01 totmalepop
+rename HC03_EST_VC01 totfemalepop
+
+xpose, clear varname format
+
+order _varname v1-v9
+
+save "D:\DCData\Libraries\RegHsg\Prog\ACS_5year_gender.dta"
```
Comment (lhendey) [inline Prog/reshape ACS data for population projection.do]: save to data folder.

**Comment by maggiesu0725** [comment]
@lhendey program is updated

File: Prog/reshape ACS data for population projection.do [hunk 9]
```diff
@@ -0,0 +1,801 @@
...
+sort Year
+
+rename HC01_EST_VC01 totpop
+rename HC02_EST_VC01 totmalepop
+rename HC03_EST_VC01 totfemalepop
+
+xpose, clear varname format
+
+order _varname v1-v11
+
```
Comment (lhendey) [inline Prog/reshape ACS data for population projection.do]: can you label these vars? for ACS_1year_gender and ACS_5Year_gender?

**Comment by maggiesu0725** [comment]
thanks, label are added @lhendey

**Comment by lhendey** [comment]
@maggiesu0725 okay to merge.

**Comment by maggiesu0725** [comment]
new program to calculate race/ethnicity for population projection is ready for review. @lhendey @rpitingolo

**Comment by rpitingolo** [comment]
@lhendey thanks. I see your comments.

**Comment by maggiesu0725** [comment]
@lhendey I confirmed with Steven and removed the group quarter restrictions. We will keep the race/ethnicity as is. Since he also wanted to project by jurisdiction I added the Jurisdiction category. Ready for finalizing?

File: Prog/Calculate population by age and raceethnicity.sas [hunk 10]
```diff
@@ -129,11 +140,11 @@ run;
 
 
 proc summary data = Race_&year. ;
```
Comment (lhendey) [inline Prog/Calculate population by age and raceethnicity.sas]: i think you need to put in a proc sort before the proc summary - I'm getting an error that the variables aren't properly sorted.

**Comment by maggiesu0725** [comment]
@lhendey sorry that I need another review. The program now add a new column for Jurisdiction. However, I discovered that 2008-2011 data is missing for Arlington, fairfax, Loudon, Alexandria and Prince Williams. From the meta data it seems like they are available? Could you let me know what possibly is the explanation?

**Comment by lhendey** [comment]
@maggiesu0725 - you can't use puma 5100600 for Loudoun in 2008-11.  It contains 3 other counties.  See in your crosswalk - it only makes up 64% of the PUMA.  There is some missing data in several jurisdictions on hispanic/all other for  80+ year olds - check with steven - maybe we need to collapse some of the top age categories?

**Comment by maggiesu0725** [comment]
@lhendey I think you were looking at the 2010 crosswalk. the 2000 crosswalk file is geocorr2014. Loudon takes up 100% of 5100600 in 2000 PUMA.

**Comment by lhendey** [comment]
@maggiesu0725 - no i'm looking at the right thing. I'm 100% certain that 5100600 contains more than loudoun county before 2012. look at the output and you can see the drop in population.

**Comment by maggiesu0725** [comment]
It's weird that the crosswalk suggests otherwise. Yes it seems like Loudoun only take up a fraction of 5100600.  Since the other two counties in 5100600 are not in the COG region, we need to take those population out. Is it OK that I use a ratio from https://usa.ipums.org/usa/volii/PUMA_composition_VA.shtml to get the total for Loudoun county? @lhendey

**Comment by lhendey** [comment]
@maggiesu0725 that would be fine for total population but not for anything by race/age. I think that its highly likely that Loudoun has a different demographic patterns than the other 3 counties. Ask Steven - but is there a way to just begin doing all the more detailed categories with it in 2012?

**Comment by maggiesu0725** [comment]
Steven suggested in that case we will be only using 2012 onwards data for Loudoun for the projection. @lhendey

**Comment by lhendey** [comment]
@maggiesu0725 sounds good.

**Comment by maggiesu0725** [comment]
Are there any changes to this program? I will close this pull request and issue. @lhendey

**Comment by lhendey** [comment]
@maggiesu0725 can you remove loudoun 2008-2011 from the output?

**Comment by maggiesu0725** [comment]
data is removed (I replaced data by missing) @lhendey

**Comment by lhendey** [comment]
@maggiesu0725 okay - please merge

**Pull request merged**